### ISSUE TITLE
Allow webpack to resolve packages from parent node_modules

### ISSUE THREAD

**Issue by eirikurn** [mozilla-neutrino/neutrino-dev#116]
Is there a reason for only using absolute paths for node_modules resolving? I.e. why `normalizePath` node_modules here:

https://github.com/mozilla-neutrino/neutrino-dev/blob/master/packages/neutrino/src/neutrino.js#L19

The webpack default is `node_modules` (relative) which triggers the commonjs logic of searching node_modules in parent directories.

In my case, I have a lerna monorepo with dependency hoisting, but webpack in the website package won't resolve dependencies from the root folder. I make this work in my neutrino preset by doing `config.resolve.modules.prepend('node_modules')`.

**Comment by eliperelman** [comment]
At the moment the default use case is a single application being generated in a project. Since the webpack dependency lives in Neutrino, there were concerns that module resolution wouldn't work correctly unless we changed from where that resolution was happening.

Or at least that was the concern when Neutrino was using webpack v1, and I can't say for sure that it would still be an issue. Do you think if we switched it from absolute to just node_modules that everything would still work as expected?

Another reason may have been when using Neutrino locally when linked, the module resolution didn't work right because node_modules wasn't in the same direct tree as the project.

**Comment by eliperelman** [comment]
Also, with the link you pointed out, this is easy to override from your project's package.json in Neutrino v5:

```json
{
  "neutrino": {
    "options": {
      "node_modules": "node_modules"
    }
  }
}
```

Please re-open if you feel this solution is insufficient. :)

**Comment by eliperelman** [comment]
Hmm, unfortunately my solution of overriding in package.json won't work since Neutrino forces it to be absolute. It think your correct solution is the way to go right now, overriding it in an override.

**Comment by eirikurn** [comment]
Yeah, it "Works on My Machine™" :)

More seriously, I think neutrino should support the resolution algorithm that developers are used to. I'm curious which modules the bundles needs to resolve from neutrino package dependencies. I can think of these cases:

* Loaders, which seem to already `require.resolve` to absolute paths.
* Extra entry points like `babel-polyfill`, `webpack-dev-server/client` and `webpack/hot/dev-server`. These can be statically resolved (like `neutrino-preset-web` does for `babel-polyfill` already) or added to `config.resolve.modules`, before or after relative `node_modules`.

**Comment by eliperelman** [comment]
Oh, why don't we just add `node_modules` in the preset in addition to the absolute one?

**Comment by eirikurn** [comment]
That's what I would suggest. Not sure if there's a reason to continue passing `<cwd>/node_modules` as options.node_modules. 

Happy to do some more testing on this if you don't have time.

**Comment by eirikurn** [comment]
Did some more testing, in a clean react project, linking local neutrino packages (not a usual setup but possibly worth supporting). This gives a resolve array like this:

```
~/projects/react-app/node_modules
~/projects/neutrino-dev/packages/neutrino-preset-web/node_modules
~/projects/neutrino-dev/packages/neutrino-preset-react/node_modules
```

This setup does not currently work, as `webpack-dev-server/client` is not resolvable, since `webpack-dev-server` is only a dependency of `neutrino`. This might be fixed by adding it as a dependency to `neutrino-preset-web`, consistent with how `webpack@2.2.1` is a dependency of both `neutrino` and `neutrino-preset-web`.

Next, I tried replacing `~/projects/react-app/node_modules` with `node_modules`. This did not work, as `react-hot-loader` (a dependency of `neutrino-preset-react`) can't find `react`, which of course *must* be singleton. Again, this is only an issue when the preset is linked. 

So if we want to support commonjs resolution with normal installs and linked installs, we need four resolve entries:

```
node_modules
~/projects/react-app/node_modules
~/projects/neutrino-dev/packages/neutrino-preset-web/node_modules
~/projects/neutrino-dev/packages/neutrino-preset-react/node_modules
```

Alternatively, if all preset dependencies are `require.resolve`'d, we could skip the latter two.

Is this something you'd be interested in a PR for?

**Comment by eliperelman** [comment]
Yes, I'd be interested in a PR. Regarding the other `node_modules`:

1. We are currently missing the relative one. This one should be in PR.
2. This one should already exist as it's defined in the Web and Node.js presets.
3. This one already exists in the Web and Node.js modules from `config.resolve.modules.add(join(__dirname, 'node_modules'))`
4. This one does not exist in the React preset, but should probably follow the same pattern as (3). Should be in PR.

Does that sound right?

**Comment by eirikurn** [comment]
Great! 

1. Yes.
2. This one is passed into the presets as a default option from `neutrino`.
3. Right.
4. The react preset already adds this.

What about the `webpack-dev-server` issue (issue with the web preset and linked neutrino)? Should I do another PR and add it as a dependency to `neutrino-preset-web`?

**Comment by eliperelman** [comment]
@eirikurn Ah you're right, the React preset does add this.

Feel free to bake these into one PR. To add the webpack dep from the root of neutrino-dev:

```bash
yarn deps:add neutrino-preset-web webpack
```

**Comment by eliperelman** [comment]
@eirikurn we plan to ship v5 this week. Do you think you will have a PR ready for this soon, or should I go ahead and take this?

**Comment by eirikurn** [comment]
Ah sorry about the delay. I was planning to finish this tomorrow (Tuesdays are open source days for me).

I definitely don't mind if you want to take over and release sooner.

**Comment by eliperelman** [comment]
@eirikurn if you want to do it on Tuesday, I am happy to wait. If something arises, please let me know. Thanks!

**Comment by eliperelman** [comment]
Fixed via #135.
